### PR TITLE
irmin-pack: Cleanups in the file stack

### PR DESCRIPTION
**Pull request by Ngoguey42**
A collection of minor changes

### CODE REVIEW

File: src/irmin-pack/unix/dispatcher.ml [hunk 1]
```diff
@@ -28,10 +28,6 @@ module Make (Fm : File_manager.S with module Io = Io.Unix) :
   module Errs = Fm.Errs
   module Control = Fm.Control
 
-  let read_suffix = ref 0
-  let read_prefix = ref 0
-  (*TODO move them in stats*)
```
Comment (metanivek) [inline src/irmin-pack/unix/dispatcher.ml:33]: Have these moved? I don't see them elsewhere this PR.

**Comment by metanivek** [comment]
Thanks for cleaning up. Left some comments/suggestions.

File: src/irmin-pack/unix/dispatcher.ml (hunk 1 above)
Comment (Ngoguey42) [reply src/irmin-pack/unix/dispatcher.ml:33]: Do they have not, but they are currently totally unused and they will be reimplemented properly when we upgrade the replay stats

File: src/irmin-pack/unix/file_manager_intf.ml [hunk 2]
```diff
@@ -20,9 +20,43 @@ module type S = sig
...
+      - 5. At the end of a GC, from the RW fiber, in [swap].
+      - 6. During integrity check routines.
+
+      1. 2. and 6. don't support SWMR and leave the store in an undefined state
+      in case of crash.
+
+      4. operates on files private to the worker. It doesn't necessitate to
+      worry about crash concistency and SWMR.
+
+      3. and 5. are highly critical. *)
```
[the remark this answers is not in the document]
Comment (Ngoguey42) [reply src/irmin-pack/unix/file_manager_intf.ml:59]: I was out of time :p I mean to write one section for each but not in this PR

File: src/irmin-pack/unix/append_only_file_intf.ml [hunk 3]
```diff
@@ -73,11 +73,11 @@ module type S = sig
 
       {3 Auto Flushes}
 
-      One of the goal of the [Append_only_file] abstraction is to provide
+      One of the goals of the [Append_only_file] abstraction is to provide
       buffered appends. [auto_flush_threshold] is the soft cap after which the
       buffer should be flushed. If a call to [append_exn] fills the buffer,
       [auto_flush_callback] will be called so that the parent abstraction takes
-      care of the flush procedure. *)
+      care of the flush procedure, which is expected to call [flush]. *)
```
[the remark this answers is not in the document]
Comment (Ngoguey42) [reply src/irmin-pack/unix/append_only_file_intf.ml:80]: Good idea. I put that in my list of cleanups

**Pull request merged**